addons/mrp/models/mrp_unbuild.py

```diff
@@ -84,11 +84,18 @@ def _onchange_company_id(self):
     @api.onchange('mo_id')
     def _onchange_mo_id(self):
         if self.mo_id:
+            if self.lot_id and self.lot_id not in self.mo_id.move_finished_ids.move_line_ids.lot_id:
+                raise UserError(_("The selected serial number does not correspond to the one used in the manufacturing order, please select another one."))
```

Comment:
Could you return a warning instead. I prefere to not block the user and warn them instead

Reply:
Done, Thanks